File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int n=nums.size(), l=0,s=0,m=INT_MAX;
        for(int r=0;r<n;r++){
            s+=nums[r];
            while(s>=target){
                m=min(m,r-l+1);
                s-=nums[l];
                l++;
            }
        }
        return (m==INT_MAX)?0:m;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l, s, m = 0, 0, float('inf')
        for r in range(len(nums)):
            s += nums[r]
            while s >= target:
                current_len = r - l + 1
                if current_len < m:
                    m = current_len
               
                s -= nums[l]
                l += 1
        return 0 if m == float('inf') else m